OpenTrons/opentrons-api
# fix(protocol-designer): update logic for ProtocolNavBar dropshadow

# Overview

We only want to show the ProtocolNavBar dropshadow if we are zoomed in on a slot, to add liquid,
hardware, or labware. This PR also refactors the logic used for text copy and alignment in the
NavBar, consolidating `hasZoomInSlot` and `isAddingHardwareOrLabware` props.

Closes RQA-3340

## Test Plan and Hands on Testing

- walk through create protocol flow and verify that navbar does not show shadow
- verify that Protocol Overview navbar does not show shadow
- click "Edit" and verify that at first, navbar does not show shadow
- verify that navbar _does_ show dropshadow when zoomed in adding/editing hardware/labware, or adding/editing liquid

## Changelog

- edit logic for navbar dropshadow
- fix screen too small z-index overlay
- fix tests

## Review requests

- see test plan

## Risk assessment

low
